src/ccd2rgb.jl

```diff
@@ -0,0 +1,40 @@
+"""
+    ccd2rgb(red::ImageHDU, green::ImageHDU, blue::ImageHDU; stretch::String="linear", shape_out = size(red))
+
+Converts 3 grayscale ImageHDU into RGB by reprojecting them.
+
+# Arguments
+- `red`: Red channel data.
+- `green`: Green channel data.
+- `blue`: Blue channel data.
+- `stretch`: Stretch function applied. Can be linear, log, sqrt or ashinh.
+- `shape_out`: Shape of output RGB image.
+"""
+function ccd2rgb(red::ImageHDU, green::ImageHDU, blue::ImageHDU; stretch::String="linear", shape_out = size(red))
+    red_rp = reproject(red, red, shape_out = shape_out)[1]
+    green_rp = reproject(green, red, shape_out = shape_out)[1]
+    blue_rp = reproject(blue, red, shape_out = shape_out)[1]
+    
+    I = (red_rp .+ green_rp .+ blue_rp) ./ 3
+    
+    if stretch == "linear"
+        I = map(x-> (x)/x , I)
+    elseif stretch == "log"
+        I = map(x-> log(x)/x , I)
+    elseif stretch == "sqrt"
+        I = map(x-> sqrt(x)/x , I)
+    elseif stretch == "asinh"
+        I = map(x-> asinh(x)/x , I)
+    else
+        throw(DomainError(stretch, "Unknown stretch function."))
+    end
+                
+    R = red_rp .* I
```

Comment:
do you need to keep `red_rp` and so on?  If not, just do the operation in-place:
```julia
red_rp .= red_rp .* I
```
You could also just replace `red_rp` with `R` above.